### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by mpsonntag**
- change DataArray deleteDimension to deleteDimensions
- fix test file issue for tag test - a change in NIX caused the order of returned tags to change, breaking the test.
- partly fixing #110 - arrays with only one row can be properly written, but more complex arrays ending with a dimension of one row still lead to an error

### CODE REVIEW

File: nix_mx.cc [hunk 1]
```diff
@@ -184,7 +184,7 @@ void mexFunction(int            nlhs,
             .reg("create_range_dimension", &nix::DataArray::createRangeDimension)
             .reg("create_alias_range_dimension", &nix::DataArray::createAliasRangeDimension)
             .reg("create_sampled_dimension", &nix::DataArray::createSampledDimension);
-        methods->add("DataArray::delete_dimension", nixdataarray::delete_dimension);
+		methods->add("DataArray::delete_dimensions", nixdataarray::delete_dimensions);
```
Comment (gicmo) [inline nix_mx.cc]: Indention change on purpose?

File: src/nixdataarray.cc [hunk 2]
```diff
@@ -52,8 +52,35 @@ namespace nixdataarray {
         nix::DataArray da = input.entity<nix::DataArray>(1);
         nix::DataType dtype = input.dtype(2);
 
-        nix::NDSize count = input.ndsize(2);
-        nix::NDSize offset(0);
+		const mxArray *arr = input.get_mx_array(2);
```
Comment (gicmo) [inline src/nixdataarray.cc]: Tabs in go, but spaces in C, please ;)

File: nix_mx.cc (hunk 1 above)
Comment (mpsonntag) [inline nix_mx.cc]: must have happened when my ms account got suspended and i had to set up vs anew. i'll change it.

File: src/nixdataarray.cc [hunk 3]
```diff
@@ -52,8 +52,35 @@ namespace nixdataarray {
...
+		bool skip = false;
+		if (dims[ndims-1] == 1) {
+			skip = true;
+			tmp = ndims - 1;
+		}
+		nix::NDSize count(tmp);
+
+		for (mwSize i = 0; i < ndims; i++) {
+			if (i != ndims - 1 || (i == ndims - 1 && !skip)) {
+				if (skip) {
```
Comment (gicmo) [inline src/nixdataarray.cc]: I must be missing something but could you just write `for (mwSize i = 0; i < tmp; i++)`?
Or maybe  just `ndims--` in line 67 and have the rest as it were?

File: src/nixdataarray.cc [hunk 4]
```diff
@@ -52,8 +52,35 @@ namespace nixdataarray {
...
+        
+        mwSize tmp = ndims;
+        
+        // Quick fix for writing data arrays that have exactly one row.
+        // This fix does not resolve more complex arrays that have a 
+        // last dimension of exactly one row.
+        bool skip = false;
+        if (dims[ndims-1] == 1) {
+            skip = true;
+            tmp = ndims - 1;
```
Comment (gicmo) [inline src/nixdataarray.cc]: Replace lines 67, 68 with `ndims--` and remove both `if` conditions.

**Pull request merged**